File: practica 1/sesion 3/DS_S3_DartFilters/GestorFiltros.dart
import 'CadenaFiltros.dart';
import 'Trabajo.dart';
import 'Filtro.dart';
import 'FiltroParteTrabajo.dart';
import 'FiltroParteTecnico.dart';

class GestorFiltros{
  CadenaFiltros cadenaFiltros;

  GestorFiltros(){
    cadenaFiltros = new CadenaFiltros();
    creaFiltroParteTrabajo();
    crearFiltroParteTecnico();
  }

  void crearFiltroParteTecnico(){
    FiltroParteTecnico technicFilter = new FiltroParteTecnico();
    cadenaFiltros.listaFiltros.add(technicFilter as Filtro);
  }

  void creaFiltroParteTrabajo(){
    FiltroParteTrabajo workFilter = new FiltroParteTrabajo();
    cadenaFiltros.listaFiltros.add(workFilter as Filtro);
  }

  void filtrar(Trabajo unTrabajo){
    cadenaFiltros.filtrar(unTrabajo);
  }
}

File: practica 1/sesion 3/DS_S3_C++Filters/GestorFiltros.cpp
#include <exception>
using namespace std;

#include "GestorFiltros.h"

GestorFiltros::GestorFiltros(){
	cadenaFiltros = new CadenaFiltros();
	creaFiltroParteTecnico();
	creaFiltroParteTrabajo();
}

void GestorFiltros::creaFiltroParteTecnico() {
	FiltroParteTecnico *technicFilter = new FiltroParteTecnico();
	cadenaFiltros->listaFiltros.push_back(technicFilter);
}

void GestorFiltros::creaFiltroParteTrabajo() {
	FiltroParteTrabajo *workFilter = new FiltroParteTrabajo();
	cadenaFiltros->listaFiltros.push_back(workFilter);
}

void GestorFiltros::filtrar(Trabajo * unTrabajo) {
	cadenaFiltros->filtrar(unTrabajo);
	// llamar desde gestor filtros al target ???
	// unTrabajo->getTecnico()->cobrar(unTrabajo->getPresupuesto());
}
